src/gsy_e/models/strategy/energy_parameters/virtual_heatpump_solver.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from dataclasses import dataclass
+from typing import Optional, List
+
+import sympy as sp
+from gsy_framework.constants_limits import ConstSettings, GlobalConfig
+from gsy_framework.utils import convert_W_to_kWh, convert_kWh_to_W
+
+from gsy_e.constants import FLOATING_POINT_TOLERANCE
+from gsy_e.models.strategy.energy_parameters.heatpump_constants import (
+    WATER_SPECIFIC_HEAT_CAPACITY,
+    WATER_DENSITY,
+    GROUND_WATER_TEMPERATURE_C,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class VirtualHeatpumpSolverParameters:
+    """Sympy solver parameters for the Heatpump."""
+
+    dh_supply_temp_C: float
+    dh_return_temp_C: float
+    dh_flow_m3_per_hour: float
+    calibration_coefficient: float = ConstSettings.HeatPumpSettings.CALIBRATION_COEFFICIENT
+    energy_kWh: Optional[float] = None
+
+    def __str__(self):
+        return (
+            f"Supply temp: {self.dh_supply_temp_C} "
+            f"Return Temp: {self.dh_return_temp_C} "
+            f"Water Flow: {self.dh_flow_m3_per_hour} "
+            f"Calibration Coefficient: {self.calibration_coefficient} "
+            f"Energy: {self.energy_kWh}"
+        )
+
+
+@dataclass
+class TankSolverParameters:
+    """Sympy solver parameters for the water tank."""
+
+    tank_volume_L: float
+    current_storage_temp_C: float
+    target_storage_temp_C: float = None
+    temp_differential_per_sec: Optional[float] = None
+
+    def __str__(self):
+        return (
+            f"Tank Volume: {self.tank_volume_L} Current Temp: {self.current_storage_temp_C}"
+            f"Target Temp: {self.target_storage_temp_C} "
+            f"Temp Differential: {self.temp_differential_per_sec}"
+        )
+
+
+class HeatpumpStorageEnergySolver:
```

Comment:
Minor, but I would like to show that this is only for the virtual HP:
```suggestion
class VirtualHeatpumpStorageEnergySolver:
```

Reply:
Agreed and fixing, thanks